### PR TITLE
Add examples 

### PR DESCRIPTION
**Pull request by diegoalzate**
# overview
adds an `examples` folder that shows how to implement the rpch sdk with `ethers`, `viem`, `wagmi` and without any other library.

### CODE REVIEW

File: examples/vanilla-sdk/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            provider: 'https://ethereum-provider.rpch.tech',
+        },
+    );
+
+    return response;
+}
+
+example()
+    .then((res) => console.log(res))
+    .catch((err) => console.error(err));
```
Comment (mjadach-iv) [inline examples/vanilla-sdk/src/index.ts:28]: For vanilla node, please change it to a async/await structure, and also print out the response with the blockNumber

**Review by mjadach-iv — CHANGES_REQUESTED** [review]

File: examples/ethers/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ */
+export class RPChProvider extends JsonRpcProvider {
+    constructor(
+        public readonly url: string,
+        public readonly sdk: SDK,
+    ) {
+        super(url);
+    }
+
+    async send(method: string, params: Array<any> | Record<string, any>): Promise<any> {
```
Comment (esterlus) [inline examples/ethers/src/index.ts:18]: isn't that the default implementation?
We don't need to override this function then, right?

File: examples/ethers/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        });
+
+        const responseJson = await rpchResponse.json();
+
+        if (JRPC.isError(responseJson)) {
+            throw new Error(responseJson.error.message);
+        }
+
+        return {
+            id: Number(responseJson.id),
```
Comment (esterlus) [inline examples/ethers/src/index.ts:69]: id does not need to be a number

File: examples/ethers/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        };
+
+        const rpchResponse = await this.sdk.send(req, {
+            provider: this.url,
+        });
+
+        const responseJson = await rpchResponse.json();
+
+        if (JRPC.isError(responseJson)) {
+            throw new Error(responseJson.error.message);
```
Comment (esterlus) [inline examples/ethers/src/index.ts:65]: do you really need to throw an error here. this is still a valid RPC response

I would expect the send function to look similar to this:

```
send(<whatever data the library gives>) {
const <data_that_sdk_understands> = magic_converter(<whatever data the library gives>);
return sdk.send(<data_that_sdk_understands>, ...)
```

File: examples/vanilla-sdk/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        },
+        {
+            provider: 'https://ethereum-provider.rpch.tech',
+        },
+    );
+
+    const responseJSON = await response.json();
+
+    if (JRPC.isError(responseJSON)) {
+        throw new Error(responseJSON.error.message);
```
Comment (esterlus) [inline examples/vanilla-sdk/src/index.ts:26]: this is a valid response and should not result in a excpetion throw

File: examples/viem/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        method: method,
+        params: params,
+    };
+
+    const rpchResponse = await sdk.send(req);
+
+    const responseJson = await rpchResponse.json();
+
+    if (JRPC.isError(responseJson)) {
+        throw new Error(responseJson.error.message);
```
Comment (esterlus) [inline examples/viem/src/index.ts:30]: not an error

File: examples/viem/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+    return responseJson.result;
+}
+
+function publicRPChClient(): PublicClient {
+    return createClient({
+        chain: mainnet,
+        transport: custom({
+            async request({ method, params }) {
+                const response = await sendRpchRequest({ method, sdk, params });
```
Comment (esterlus) [inline examples/viem/src/index.ts:41]: you can try this:

`request({ method, params }) => sendRpchRequest({ method, sdk, params })`

would be nice if it works

File: examples/wagmi/src/App.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  };
+
+  const rpchResponse = await sdk.send(req);
+
+  console.log("rpchResponse", rpchResponse);
+
+  const responseJson = await rpchResponse.json();
+
+  if (JRPC.isError(responseJson)) {
+    throw new Error(responseJson.error.message);
```
Comment (esterlus) [inline examples/wagmi/src/App.tsx:40]: not an error

**Review by esterlus — CHANGES_REQUESTED** [review]
Great work,

Unfortunately the eslint-plugin-compat does not work with typescript.
I am going to remove it from sdk and compatcrypto. feel free to remove it here as well

File: examples/ethers/src/index.ts (hunk 2 above)
Comment (diegoalzate) [reply examples/ethers/src/index.ts:18]: this is correct, going to remove it.

File: examples/ethers/src/index.ts (hunk 3 above)
Comment (diegoalzate) [reply examples/ethers/src/index.ts:69]: for ethers, their response type expects a number:
```

/**
 *  A JSON-RPC result, which are returned on success from a JSON-RPC server.
 */
export type JsonRpcResult = {
    /**
     *  The response ID to match it to the relevant request.
     */
    id: number;

    /**
     *  The response result.
     */
    result: any;
};

```

File: examples/viem/src/index.ts (hunk 7 above)
Comment (diegoalzate) [reply examples/viem/src/index.ts:41]: removed `sendRpchRequest` abstraction

File: examples/vanilla-sdk/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    return responseJSON;
+}
+
+async function main() {
+    try {
+        const response = await example();
+        console.log(response);
+    } catch (e) {
+        console.error(e);
+    }
```
Comment (esterlus) [inline examples/vanilla-sdk/src/index.ts:34]: can be simplified: 
`example().then(console.log).catch(console.error);`

**Review by esterlus — APPROVED** [review]
Great work

**Pull request merged**